### ISSUE TITLE
Copy&Paste problem

### ISSUE THREAD

**Issue by massimobiagioli** [cedrozor/myrtille#192]
Hi,
When you press the **CTRL + V** key combination (to paste a text into the remote app launched by Myrtille) the current session is disconnected.
The **CTRL + C** key combination works correctly (remote clipboard synchronization)

The version used is **2.7.0**

Here is the error reported in the log file:

```
2019-12-12 11: 48: 59.712 [207] ERROR System.Diagnostics redirection [(null)] - Failed to process input (s) MMO306-8, remote session 74ef8114-262c-43da-baa7-b18a653fc424 (System.ObjectDisposedException : The CancellationTokenSource has been disposed.
   at System.Threading.CancellationTokenSource.ThrowObjectDisposedException ()
   at Myrtille.Web.RemoteSessionManager.ProcessInputs (HttpSessionState session, String data))
2019-12-13 09:41: 26,098 [37] ERROR System.Diagnostics redirection [(null)] - Failed to retrieve the active remote session (System.NullReferenceException: Object reference not set to an instance of an object.
   at Myrtille.Web.SendInputs.Page_Load (Object sender, EventArgs e))
```

**Comment by cedrozor** [comment]
Hi, thanks for the report. Is there a specific clipboard content (with special characters perhaps) that triggers the disconnect?

The issue is more on the freerdp side (the rdp client used by myrtille). You can enable the freerdp logs by setting the following key to "true" into Myrtille.Services.exe.config (located into the myrtille install path, bin folder):

```
<!-- log remote session events into a file (located into <Myrtille folder>\log) -->
<add key="RemoteSessionLog" value="true" />
```

Then restart the windows service "Myrtille.Services". I'm interested by the logs preceding the freerdp crash.

**Comment by camjcorley** [comment]
Hey cedrozor, I was able to replicate the issue on 2.7.1:

IE 11 & Firefox 71: 
- Ctrl+v crashes the connection if the remote clipboard has not been set. So this only happens until you copy something within the session. 
- Attempting to paste via context menu from within the session (right-click > paste) does not successfully read the local clipboard.

This issue does not occur in Google Chrome 79.

I cleared the log files then started the service and replicated the issue (attached).
Let me know if there's anything else I can do to help gather additional info.

[wfreerdp.15708.log](https://github.com/cedrozor/myrtille/files/3976564/wfreerdp.15708.log)
[Myrtille.Services.log](https://github.com/cedrozor/myrtille/files/3976565/Myrtille.Services.log)
